src/Data/YAML.hs

```diff
@@ -531,3 +497,80 @@ decode1Strict :: FromYAML v => BS.ByteString -> Either String v
 decode1Strict text = do
   vs <- decodeStrict text
   maybe (Left "expected unique") Right $ listToMaybe vs
+
+
+
+-- | A type from which YAML nodes can be constructed
+--
+-- @since 0.2.0.0
+class ToYAML a where
+  toYAML :: a -> Node () -- ^ Convert a Haskell value to a YAML Node data type.
+
+-- | Trivial instance
+instance ToYAML (Node ()) where
+  toYAML = id
+
+instance ToYAML (Node Pos) where
+  toYAML node = case node of
+      Scalar   _ scalar -> Scalar () scalar
+      Mapping  _ tag m  -> Mapping () tag (Map.fromList $ map (\(k,v) -> (toYAML k , toYAML v)) (Map.toList m)) 
+      Sequence _ tag s  -> Sequence () tag (map toYAML s)
+      Anchor   _ n nod  -> Anchor () n (toYAML nod)
+
+instance ToYAML Bool where
+  toYAML = Scalar () . SBool
+
+instance ToYAML Double where
+  toYAML = Scalar () . SFloat
+
+instance ToYAML Int     where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Int8    where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Int16   where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Int32   where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Int64   where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Word    where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Word8   where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Word16  where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Word32  where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Word64  where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Natural where toYAML = Scalar () . SInt . toInteger
+instance ToYAML Integer where toYAML = Scalar () . SInt
+
+
+instance ToYAML Text where
+  toYAML = Scalar () . SStr
+
+instance ToYAML a => ToYAML (Maybe a) where
+  toYAML Nothing = Scalar () SNull
+  toYAML (Just a) = toYAML a
+
+instance (ToYAML a, ToYAML b) => ToYAML (Either a b) where
```

Comment:
I'd rather not define this instance as it's tricky to round-trip (i.e. we don't have a proper `FromYAML` instance for that either)